### PR TITLE
Chenges required for the new project model

### PR DESCRIPTION
**Pull request by slluis**

### CODE REVIEW

File: MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig [hunk 1]
```diff
@@ -45,33 +45,8 @@
...
-    <!--  If FSharp 3.1 is available, use it. Note XS looks for the first DotNetProject node under 'Extension' -->
-	<Condition id="MSBuildTargetIsAvailable" target="$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.1\Framework\v4.0\Microsoft.FSharp.Targets" >
-      <DotNetProject language="F#" extension="fsproj" guid="{f2a71f9b-5d33-465a-a702-920d77279786}" import="$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.1\Framework\v4.0\Microsoft.FSharp.Targets"  resourceHandler="MonoDevelop.FSharp.FSharpResourceIdBuilder"/>
-	</Condition>
-
-    <!--  If FSharp 3.1 is not available, then use FSharp 3.0  -->
-    <DotNetProject language="F#" extension="fsproj" guid="{f2a71f9b-5d33-465a-a702-920d77279786}" import="$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.0\Framework\v4.0\Microsoft.FSharp.Targets"  resourceHandler="MonoDevelop.FSharp.FSharpResourceIdBuilder"/>
-
-    <!-- Disable this until the new project system is available
-    <Condition id="MSBuildTargetIsAvailable" target="$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.1\Framework\v4.0\Microsoft.Portable.FSharp.Targets">
```
Comment (7sharp9) [inline MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig]: How will handling adding: `$(MSBuildExtensionsPath32)\..\Microsoft SDKs\F#\3.1\Framework\v4.0\Microsoft.Portable.FSharp.Targets`
work for F# pcl projects?

File: MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig [hunk 2]
```diff
@@ -86,9 +61,10 @@
     <Parser class = "MonoDevelop.FSharp.FSharpFoldingParser" mimeType="text/x-fsharp" />
   </Extension>
 
-  <Extension path = "/MonoDevelop/SourceEditor2/TooltipProviders">
+<!-- TODO NPM type loading fails for some reason -->
+<!--  <Extension path = "/MonoDevelop/SourceEditor2/TooltipProviders">
```
Comment (7sharp9) [inline MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig]: Does this mean tooltips are currently unavailable?

File: MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig (hunk 1 above)
Comment (slluis) [inline MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig]: It has to be handled here: https://github.com/fsharp/FSharpMDXS/pull/12/files#diff-8f34e9f5308564680d2190ae8c0622e1R32

File: MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig (hunk 2 above)
Comment (slluis) [inline MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig]: Yes, I'll take a look at this now.

File: MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig (hunk 1 above)
Comment (7sharp9) [inline MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig]: @slluis Im not sure I follow, how would that function know that it is a portable project?

Or do you mean I have to create another subclass of FSharpProject that is referenced in the portable templates?

Comment (slluis) [inline MonoDevelop.FSharpBinding/FSharpBinding.addin.xml.orig]: No, you don't need another class. On FSharpProject you can override OnInitializeFromTemplate(). In that method you are given the Options xml element of the template, so you can check if the template is for a portable library. You can then flag the project as being a portable library (using a boolean field for example), and when OnGetDefaultImports is called, you can check that flag and return the portable imports.

**Pull request merged**